autoload/dirvish.vim

```diff
@@ -42,21 +42,21 @@ function! s:parent_dir(dir) abort
 endfunction
 
 if v:version > 703
-function! s:globlist(pat) abort
-  return glob(a:pat, !s:suf(), 1)
+function! s:globlist(dir_esc, pat) abort
+  return globpath(a:dir_esc, a:pat, !s:suf(), 1)
 endfunction
 else "Vim 7.3 glob() cannot handle filenames containing newlines.
-function! s:globlist(pat) abort
-  return split(glob(a:pat, !s:suf()), "\n")
+function! s:globlist(dir_esc, pat) abort
+  return split(globpath(a:dir_esc, a:pat, !s:suf()), "\n")
 endfunction
 endif
 
 function! s:list_dir(dir) abort
-  " Escape for glob().
-  let dir_esc = escape(substitute(a:dir,'\[','[[]','g'),'{}')
-  let paths = s:globlist(dir_esc.'*')
+  " Escape comma for globpath().
+  let dir_esc = escape(a:dir, ',')
+  let paths = s:globlist(dir_esc, '*')
   "Append dot-prefixed files. glob() cannot do both in 1 pass.
```

Comment:
can we avoid 2 passes now? will this work:

    let paths = s:globlist(dir_esc, '*,.[^.]*')